src/System.Private.CoreLib/shared/System/Number.Formatting.cs

```diff
@@ -391,57 +391,18 @@ private static string FormatDouble(ref ValueStringBuilder sb, double value, Read
             NumberBuffer number = default;
             number.kind = NumberBufferKind.Double;
 
-            switch (fmt)
+            if ((fmt == 'R') || (fmt == 'r'))
             {
-                case 'R':
-                case 'r':
-                    {
-                        // In order to give numbers that are both friendly to display and round-trippable, we parse the
-                        // number using 15 digits and then determine if it round trips to the same value. If it does, we
-                        // convert that NUMBER to a string, otherwise we reparse using 17 digits and display that.
-                        DoubleToNumber(value, DoublePrecision, ref number);
-                        if (number.scale == ScaleNAN)
-                        {
-                            return info.NaNSymbol;
-                        }
-                        else if (number.scale == ScaleINF)
-                        {
-                            return number.sign ? info.NegativeInfinitySymbol : info.PositiveInfinitySymbol;
-                        }
-
-                        if (NumberToDouble(ref number) == value)
-                        {
-                            NumberToString(ref sb, ref number, 'G', DoublePrecision, info);
-                        }
-                        else
-                        {
-                            DoubleToNumber(value, 17, ref number);
-                            NumberToString(ref sb, ref number, 'G', 17, info);
-                        }
-
-                        return null;
-                    }
-
-                case 'E':
-                case 'e':
-                    // Round values less than E14 to 15 digits
-                    if (digits > 14)
-                    {
-                        precision = 17;
-                    }
-                    break;
-
-                case 'G':
-                case 'g':
-                    // Round values less than G15 to 15 digits. G16 and G17 will not be touched.
-                    if (digits > 15)
-                    {
-                        precision = 17;
-                    }
-                    break;
+                fmt = 'G';
+                digits = DoublePrecision;
+            }
+            else if (digits > 0)
```

Comment:
`G` results in `digits = -1` and `G0` results in `digits = 0`.

In both cases, this results in the default precision (`17` for double and `9` for single).

For `G0`, this matches the current behavior. The IEEE spec indicates that the lower limit for significant digits converted should be 1. So, we could also just make `G0` be treated as `G1`.
* Thoughts/Opinions?